t/compare_movements_spec.vim

```diff
@@ -0,0 +1,272 @@
+            let l:list += [l:curpos]
+            let l:lastpos = l:curpos
+            call TryNormal('zz')
+            call TryNormal(l:handler)
+            let l:curpos = getpos(".")[1:2]
+            " Abort after a fixed number of steps.
+            if len(l:list) > s:maximal_number_of_compared_movments
+                break
+            endif
+        endwhile
+    endfor
+    " }}
+
+    " The resulting lists are stored in l:jumpmarks[*][1], now.
+    let [l:cursor_positions1, l:cursor_positions2] = [ l:jumpmarks[0][1], l:jumpmarks[1][1] ]
+
+    if l:cursor_positions1 == l:cursor_positions2
+        return 0
+    endif
+
+    " Search for first unmatching position. {{
+    let l:index = 0
+    let l:len = min([len(l:cursor_positions2), len(l:cursor_positions1)]) 
+    while l:index < l:len
+        Expect l:cursor_positions2[l:index] to_cursor l:cursor_positions1[l:index]
+        let l:index += 1
+    endwhile
+
+    " Collision with begin or end of file or while loop aborts to early.
+    if a:backward == 1
+        Expect join([a:movement2, ': File begin reached after ', len(l:cursor_positions2), ' steps.'])
+                    \ == join([a:movement1, ': File begin reached after ', len(l:cursor_positions1), ' steps.'])
+    else
+        Expect l:cursor_positions2[l:index-1] to_cursor l:cursor_positions1[l:index]
+        Expect join([a:movement2, ': File end reached after ', len(l:cursor_positions2), ' steps.'])
+                    \ == join([a:movement1, ': File end reached after ', len(l:cursor_positions1), ' steps.'])
+    endif
+    " }}
+
+    return -1
+endfunction
+
+" Hand crafted text with rare cases
+function! InsertTestText1()
+
+    " Blanks at document begin
+    call AddLine('')
+    call AddLine(' ')
+    call AddLine('')
+
+    call AddLine('scriptencoding utf-8')
+    
+    " '^\s*[not-\k]'-case
+    call AddLine('!foo')
+    call AddLine('   !bar')
+
+    call AddLine('<!{}>s!  ')
+
+    " Blanks at document end
+    call AddLine('')
+    call AddLine(' ')
+    call AddLine('')
+endfunction
+
+"}}}
+
+"Keyword word motion {{{
+describe 'Keyword word motion'
+    before
+        new
+        resize 10
+        nmap a <Nop>
+        let g:EasyMotion_keys = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+        let g:EasyMotion_maximal_jumpmarks  = 2 " Error for value 1 unanalyzed. 
+        nmap <Leader>w <Plug>(easymotion-iskeyword-w)
+        nmap <Leader>b <Plug>(easymotion-iskeyword-b)
+        nmap <Leader>e <Plug>(easymotion-iskeyword-e)
+        nmap <Leader>ge <Plug>(easymotion-iskeyword-ge)
+        nmap <Leader>W <Plug>(easymotion-W)
+        nmap <Leader>B <Plug>(easymotion-B)
+        nmap <Leader>E <Plug>(easymotion-E)
+        nmap <Leader>gE <Plug>(easymotion-gE)
+        call EasyMotion#init()
+        call vspec#customize_matcher('to_cursor', s:to_cursor)
+    end
+
+    after
+        close!
+    end
+
+    it 'Simple test to check setup of this test'
+        normal aa\<Esc>
```

Comment:
If EasyMotion enable the overlay for the jumping it requests for the target (like \\wa), but if only one target is reachable the 'a' is not required.
Then, a starts the insert mode. I deliberate how to fix that and decide to remap 'a' in normal mode during the test.
The line normal aa\<Esc> with the following Except should test if the remapping was successful/active. I will add a comment.